### ISSUE TITLE
MDBOOK010 incorrectly triggers on dollar signs inside code blocks

### ISSUE THREAD

**Issue by joshrotenberg** [joshrotenberg/mdbook-lint#141]
### Description
The MDBOOK010 rule (math block validation) is incorrectly triggering on dollar signs (`$`) inside fenced code blocks. These dollar signs are legitimate bash variables, not math delimiters.

### Steps to Reproduce

1. Create a markdown file with the following content:
```markdown
```bash
echo "Database status: $status"
if [ "$status" = "active" ]; then
    echo "Active"
fi
```
```

2. Run `mdbook-lint lint file.md`

### Expected Behavior
Code blocks should be isolated from math delimiter detection. Dollar signs inside code blocks should not trigger MDBOOK010.

### Actual Behavior
MDBOOK010 reports: "Unclosed inline math block" for bash variables inside code blocks.

### Impact
This creates numerous false positives in technical documentation that includes shell scripts, making it difficult to use mdbook-lint with documentation that contains bash/shell examples.

### Suggested Fix
The math block detection should skip content inside fenced code blocks entirely. The rule should only apply to inline and block math in regular markdown content.

### Environment
- mdbook-lint version: 0.11.1
- Operating System: macOS

### Additional Context
This issue was reported by a user setting up mdbook-lint for technical documentation with many shell script examples. The false positives made the tool difficult to use for this use case.
